src/main/java/net/dv8tion/jda/api/entities/MessageType.java

```diff
@@ -148,25 +148,31 @@
      *
      * Messages from this type usually come with custom embeds containing relevant information, the author is the user that triggered the filter.
      */
-    AUTO_MODERATION_ACTION(24),
+    AUTO_MODERATION_ACTION(24, true, true),
 
     /**
      * Unknown MessageType.
      */
-    UNKNOWN(-1);
+    UNKNOWN(-1, false, true);
 
     private final int id;
     private final boolean system;
+    private final boolean deletable;
 
-    MessageType(int id)
-    {
-        this(id, true); // true as default because the majority are system
-    }
-
-    MessageType(int id, boolean system)
-    {
+    MessageType(int id, boolean system, boolean deletable) {
         this.id = id;
         this.system = system;
+        this.deletable = deletable;
+    }
+
+    MessageType(int id, boolean system) {
+        // most system messages are not deletable, most non-system messages are
```

Comment:
Pretty much every system message is deletable, the only one i know that prevents it is automod. Do you have actual exampels? I'm not a fan of having to say "true, true" on every enum constant here.

Reply:
Outside of Automod, Message sent in group DMs when changing the icon or the name, making calls, addition or removal of members are not deletable. But that's that

Reply:
Group DMs are not even accessible to bots normally, so we shouldn't really worry about those types at all. We could comment them out instead. From what i can tell the only actually undeletable messages accessible to bots, are thread start messages and automod, am i wrong?

Reply:
Automod messages can be deleted. The two that can't be deleted are channel name and thread starter.

Reply:
Then they changed it, they were not deletable when the type was added.

Reply:
Yes, that is what this PR is primarily addressing. 
![image](https://user-images.githubusercontent.com/16168171/174501100-17315ae8-94b8-46da-b26b-d13c531965e3.png)